server.coffee

```diff
@@ -462,31 +483,41 @@ app.post '/reset/:token', (req, res) ->
           'This is a confirmation that the password for your account ' + user.email + ' has just been changed.\n'
       }
       smtpTransport.sendMail mailOptions, (err) ->
-        #req.flash 'success', 'Success! Your password has been changed.'
+        # req.flash 'success', 'Success! Your password has been changed.'
         throw err if err
         done(err)
   ], (err) ->
     throw err if err
     res.redirect('/')
 
+app.post '/update-profile', (req, res) ->
+  if req.user
+    db.collection('users').update {username: req.user.username}, {$set: {options: {background: req.body.background,\
+      'alt-arts': req.body['alt-arts'], 'opponent-alt-art': req.body['opponent-alt-art']}}}, \
+      (err) ->
+        console.log(err) if err
+        res.send {message: 'OK', background: req.body.background, altarts: req.body['alt-arts']}, 200
+  else
+    res.send {message: 'Unauthorized'}, 401
```

Comment:
This is the old style. Needs to be `res.status(401).send({message: 'Unauthorized'})`. Same for L499 (although the default status is 200).